src/nft/components/layout/Checkbox.tsx

```diff
@@ -0,0 +1,44 @@
+import clsx from 'clsx'
+import { Box } from 'nft/components/Box'
+import { ApprovedCheckmarkIcon } from 'nft/components/icons'
+import { ChangeEvent } from 'react'
+
+import * as styles from './Checkbox.css'
+
+interface CheckboxProps {
+  checked: boolean
+  hovered: boolean
+  children: React.ReactNode
+  onChange: (e: ChangeEvent<HTMLInputElement>) => void
+}
+
+export const Checkbox = ({ onChange, children, checked, hovered, ...props }: CheckboxProps) => {
```

Comment:
this really shouldn't have `onChange` or `checked` in `CheckboxProps` it can come from `HTMLInputElement` props that you can extend from either on the function arg typing or `CheckboxProps` since you they are being passed in `{...props}` in the `input` component